core/src/main/scala/scala/collection/parallel/Tasks.scala

```diff
@@ -180,15 +180,16 @@ trait AdaptiveWorkStealingTasks extends Tasks {
     def spawnSubtasks() = {
       var last: AWSTWrappedTask[R, Tp] = null
       var head: AWSTWrappedTask[R, Tp] = this
-      do {
+      while ({
         val subtasks = head.split
         head = subtasks.head
         for (t <- subtasks.tail.reverse) {
           t.next = last
           last = t
           t.start()
         }
-      } while (head.body.shouldSplitFurther)
+        head.body.shouldSplitFurther
```

Comment:
it really frustrates me that we're doing away with `do { ... } while (...)` and ending up with... this. but that's not really a comment on this PR